### PR TITLE
Fixes a network problem

### PR DESCRIPTION
**Pull request by ehankland**
Since networks were shared across all benchmark runs, this could lead to one benchmark attempting to delete it while it was still in use

Also make the the way we open ports on the firewall compatible with having VMs from different clouds in the same benchmark

### CODE REVIEW

File: perfkitbenchmarker/aws/aws_virtual_machine.py [hunk 1]
```diff
@@ -112,16 +111,15 @@ class AwsVirtualMachine(virtual_machine.BaseVirtualMachine):
   imported_keyfile_set = set()
   deleted_keyfile_set = set()
 
-  def __init__(self, vm_spec):
+  def __init__(self, vm_spec, network, firewall):
     """Initialize a AWS virtual machine.
 
     Args:
       vm_spec: virtual_machine.BaseVirtualMachineSpec object of the vm.
```
Comment (yuyantingzero) [inline perfkitbenchmarker/aws/aws_virtual_machine.py]: Add documentations for network and firewall.

Comment (ehankland) [inline perfkitbenchmarker/aws/aws_virtual_machine.py]: Done

Comment (ehankland) [inline perfkitbenchmarker/aws/aws_virtual_machine.py]: oops did this only on BaseVirtualMachine - I'll do it for all of them - will update in a minute

Comment (ehankland) [inline perfkitbenchmarker/aws/aws_virtual_machine.py]: Done

File: perfkitbenchmarker/gcp/gce_virtual_machine.py [hunk 2]
```diff
@@ -261,8 +261,8 @@ class WindowsGceVirtualMachine(GceVirtualMachine,
   BOOT_DISK_SIZE_GB = 50
   BOOT_DISK_TYPE = disk.REMOTE_SSD
 
-  def __init__(self, vm_spec):
-    super(WindowsGceVirtualMachine, self).__init__(vm_spec)
+  def __init__(self, vm_spec, network, firewall):
+    super(WindowsGceVirtualMachine, self).__init__(vm_spec, network, firewall)
```
Comment (yuyantingzero) [inline perfkitbenchmarker/gcp/gce_virtual_machine.py]: Documentation missing here.

File: perfkitbenchmarker/kubernetes/kubernetes_virtual_machine.py [hunk 3]
```diff
@@ -57,8 +57,8 @@ class KubernetesVirtualMachine(virtual_machine.BaseVirtualMachine):
   Object representing a Kubernetes POD.
   """
 
-  def __init__(self, vm_spec):
-    super(KubernetesVirtualMachine, self).__init__(vm_spec)
+  def __init__(self, vm_spec, network, firewall):
```
Comment (yuyantingzero) [inline perfkitbenchmarker/kubernetes/kubernetes_virtual_machine.py]: Documentation missing here.

File: perfkitbenchmarker/openstack/os_virtual_machine.py [hunk 4]
```diff
@@ -51,8 +51,9 @@ class OpenStackVirtualMachine(virtual_machine.BaseVirtualMachine):
     # Subclasses should override the default image.
     DEFAULT_IMAGE = None
 
-    def __init__(self, vm_spec):
-        super(OpenStackVirtualMachine, self).__init__(vm_spec)
+    def __init__(self, vm_spec, network, firewall):
```
Comment (yuyantingzero) [inline perfkitbenchmarker/openstack/os_virtual_machine.py]: Documentation missing here.

**Comment by yuyantingzero** [comment]
A few small comments, LGTM

**Comment by ehankland** [comment]
Fixed

**Pull request merged**